### PR TITLE
Added logserver feature flag

### PR DESCRIPTION
**Pull request by olaaun**

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/DomAdminV4Builder.java [hunk 1]
```diff
@@ -100,8 +102,7 @@ private NodesSpecification createNodesSpecificationForLogserver() {
         if (deployState.getProperties().useDedicatedNodeForLogserver() &&
                 context.getApplicationType() == ConfigModelContext.ApplicationType.DEFAULT &&
                 deployState.isHosted() &&
-                deployState.zone().system() == SystemName.cd &&
-                deployState.zone().environment() == Environment.dev)
+                Flags.ENABLE_LOGSERVER.bindTo(deployState.flagSource()).value())
```
Comment (hakonhall) [inline config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/DomAdminV4Builder.java]: In order to get the flag value for a particular application, you need to tell the flag what application this is for before value() with a call to `with(Dimension.APPLICATION_ID, anApplicationId.serializedForm())`

**Comment by hakonhall** [comment]
FYI @baldersheim

**Review by hmusum — APPROVED** [review]
LGTM

**Review by hakonhall — APPROVED** [review]

**Comment by hmusum** [comment]
Config server does not start now, I am seeing:

`[16:19:15.996] [2019-01-10 16:18:53.865] WARNING : configserver     stderr	Caused by: java.lang.IllegalStateException: No global component of class com.yahoo.vespa.flags.FlagSource to inject into component 'VespaModelFactory' of type 'com.yahoo.vespa.model.VespaModelFactory'.`

I need to run now, but it would be nice if someone can revert/look into this

**Comment by hakonhall** [comment]
I'll look into it and additional issues on the controller in cd

**Comment by hakonhall** [comment]
I assume the root cause is not being able to create ConfigServerFlagSource.
Fix in https://github.com/vespa-engine/vespa/pull/8094

**Comment by hakonhall** [comment]
That didn't improve much - debugging further

**Comment by hakonhall** [comment]
Found this for controller:

```
	Failed to set up first component graph
	exception=
	java.lang.RuntimeException: When resolving dependencies of 'VespaModelFactory.6.329.34' of type 'com.yahoo.vespa.model.VespaModelFactory'
	Caused by: java.lang.IllegalStateException: No global component of class com.yahoo.vespa.flags.FlagSource to inject into component 'VespaModelFactory.6.329.34' of type 'com.yahoo.vespa.model.VespaModelFactory'.
```

The presence of a version number here indicates and reminds me that there may be some special component graph setting up in the config model. It's probably best to roll this back and discuss this with Harald and Gjøran tomorrow.

**Comment by hakonhall** [comment]
For each version, there is a single JAR file/bundle (vespa-model-fat-amended) with ~4 components loaded from it, among them VespaModelFactory.

6.329.34 contains the fix to ConfigServerFlagSource. Once no config models contains a Vespa version older than this in CD/prod, this PR can be retried. @hmusum where's the best place to check the list of versions?

**Pull request merged**